### PR TITLE
Week 5 Assignments completed

### PR DESCRIPTION
**Pull request by joeshajan**
## Exercises
- exercise 5.1
- exercise 5.2
- exercise 5.3
- exercise 5.4

## Description
- Exercise using async/await and generators.
- Exercise for counting vowels in a string and returning the map.
- Exercise to find if any duplicate values are present in an array using a set.
- Exercise to test function using jest.

## Type of Change
  
- [x] assignments completed

## Where should the reviewer start? :checkered_flag:
(Explain where the reviewer should start in order to review the whole addition or subtractions)

### Exercise 5.1

[link to exercise 5.1](https://github.com/pesto-students/p7-rahulm-joeshajan/blob/week5/Week-5/exercise%205.1/index.js)

**output**

image

### Exercise 5.2

[link to exercise 5.2](https://github.com/pesto-students/p7-rahulm-joeshajan/blob/week5/Week-5/exercise%205.2/index.js)

**output**

image

### Exercise 5.3

[link to exercise 5.3](https://github.com/pesto-students/p7-rahulm-joeshajan/blob/week5/Week-5/exercise%205.3/index.js)

**output**

image

### Exercise 5.4

[link to exercise 5.4 index file](https://github.com/pesto-students/p7-rahulm-joeshajan/blob/week5/Week-5/exercise%205.4/index.js)

[link to exercise 5.4 test file](https://github.com/pesto-students/p7-rahulm-joeshajan/blob/week5/Week-5/exercise%205.4/calculator.test.js)

**output**

image

### CODE REVIEW

File: Week-5/exercise 5.2/index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const isVowel = (char) => {
```
Comment (rahulm2) [inline Week-5/exercise 5.2/index.js:1]: @joeshajan we dont need to use isVowel. Check the guidelines again. We need a map with keys as numbers. Try to implement it w/o isVowel

File: Week-5/exercise 5.3/index.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const hasDuplicate = (arr) => new Set(arr).size !== arr.length;
```
Comment (rahulm2) [inline Week-5/exercise 5.3/index.js:1]: do we really need to create the whole set? Dont you think we can decide early? This logic will always be best case O(n)

File: Week-5/exercise 5.4/calculator.test.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const mathOperators = require("./index");
+
+test("properly adds two numbers", () => {
```
Comment (rahulm2) [inline Week-5/exercise 5.4/calculator.test.js:3]: Do it using BDD approach

File: Week-5/exercise 5.2/index.js (hunk 1 above)
Comment (joeshajan) [reply Week-5/exercise 5.2/index.js:1]: Hey @rahulm2 sorry I don't get what you said.

Is this the output should be?

image

**Review by rahulm2 — APPROVED** [review]
lgtm.

**Pull request merged**